library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/recorder/Node.kt

```diff
@@ -6,4 +6,10 @@
 
 package com.datadog.android.sessionreplay.recorder
 
-internal data class Node(val id: Int, val parent: Node? = null)
+import com.datadog.android.sessionreplay.model.MobileSegment
+
+internal data class Node(
+    val id: Long,
+    val wireframes: List<MobileSegment.Wireframe>,
```

Comment:
suggestion: I think it might be healthier to not depend directly on (HTTP) transport details in the core SR logic (the `MobileSegment.Wireframe` is what we serialize and send in payload). Introducing intermediate (DTO) and doing `Node` → indermediate → `Wireframe` transformation could IMO help on a long run by clearly separating concerns. Otherwise, the Recorder code will have to be changed if transport details change, which seems to be against SoC as both layers clearly overlap here. WDYT?

Reply:
Yes I thought about it but then I think I prefer it this way even though maybe at some point I need to change the wireframes a bit. Main reason why is:
- I don't want to introduce another layer which will require a transformer. If there's any change on the Wireframe I will still have to apply it at the transformer level from the intermediate DTO to Wireframe
- Most important I want to avoid creating too many objects. With this approach the wireframes instances created here will be re - used at the Processor level.